src/board/square.rs

```diff
@@ -0,0 +1,59 @@
+use super::{Board, errors::InvalidSquareError, Orientation};
+
+// Represents a square on the board
+//
+// File counts from the left, starts at 0
+// Rank counts from the bottom, starts at 0
+#[derive(Copy, Clone, Debug, PartialEq, Eq)]
+pub struct Square {
+    pub file: usize,
+    pub rank: usize,
+}
+
+impl Square {
+    pub fn get_file(&self) -> usize {
+        return self.file;
+    }
+
+    pub fn get_rank(&self) -> usize {
+        return self.rank;
+    }
+}
+
+// Represents a possibly invalid location of a square
+// on the board.
+// File counts from the left, starts at 0
+// Rank counts from the bottom, starts at 0
+#[derive(Copy, Clone, Debug, PartialEq, Eq)]
+pub struct UncheckedSquare {
+    pub file: usize,
```

Comment:
Could the `file` and `rank` not be `isize`s? This way, we could also share the
code for validations for going under 0, e.g. with offsetes.

Reply:
That seems like a good idea, should I do that in this PR or wait since its kinda a different problem?